Userland/Utilities/slugify.cpp

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2023, Gurkirat Singh <tbhaxor@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/CharacterTypes.h>
+#include <AK/Error.h>
+#include <AK/Slugify.h>
+#include <AK/String.h>
+#include <AK/Vector.h>
+#include <LibCore/ArgsParser.h>
+#include <LibMain/Main.h>
+#include <LibUnicode/Normalize.h>
+
+ErrorOr<int> serenity_main(Main::Arguments arguments)
+{
+    Vector<String> inputs;
+    String output_type;
+    char glue = '-';
+    bool spa;
+
+    Core::ArgsParser parser;
+    parser.set_general_help("Slugify is a simple text to slug transform utility\n$ slugify 'Serenity is a cool ### PROject123.'");
+    parser.add_option(output_type, "Output format to choose from 'md', 'html', 'plain'. (default: md)", "format", 'f', "FORMAT");
+    parser.add_option(Core::ArgsParser::Option {
+        .help_string = "Specify delimiter to join the parts. (default: -)",
+        .long_name = "glue",
+        .short_name = 'g',
+        .value_name = "GLUE",
+        .accept_value = [&glue](StringView s) -> ErrorOr<bool> {
+            if (s.length() > 0) {
+                glue = s.to_deprecated_string()[0];
+            }
+            return true;
```

Comment:
We only accept a single ascii character, so reject any input that isn't. There's also no need to convert to a DeprecatedString.

```suggestion
            if (s.length() == 1 && is_ascii_printable(s[0])) {
                glue = s[0];
                return true;
            }
            return false;
```

(And then also mention this in the help string.)

Reply:
Yeah, I thought string is not sub-scriptable. Also if it is not a valid, I am planning to use the default value `-`. So I don't think so return `false` is required.

Reply:
If what the user passed in isn't valid, you should return false. Otherwise it'll look like it accepts multi-character strings but only use the first character, which is confusing. It's better for the program to just reject the input, print the help message, and the user can correct it and try again.

Reply:
I see. So this is basically to make it explicit that the argument accept single character input only.